### ISSUE TITLE
feat: add blanks-around-headings rule (MD022)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#101]
## Overview

Implement the `blanks-around-headings` rule (equivalent to markdownlint MD022).

Closes part of #76.

## Rule Description

Headings must be surrounded by blank lines.

- There must be a blank line **before** a heading (except at the start of the document or after frontmatter)
- There must be a blank line **after** a heading (except at the end of the document)

## Example

**Invalid:**
```markdown
Some text
## Heading
More text
```

**Valid:**
```markdown
Some text

## Heading

More text
```

## Implementation Notes

- Rule key: `blanks-around-headings`
- Config field: `BlanksAroundHeadings bool`
- Follows standard rule implementation checklist
